MatrixSDK/MXSession.m

```diff
@@ -460,6 +669,8 @@ - (void)startWithSyncFilter:(MXFilterJSONModel*)syncFilter
 
 - (void)startWithSyncFilterId:(NSString *)syncFilterId onServerSyncDone:(void (^)(void))onServerSyncDone failure:(void (^)(NSError *))failure
 {
+    [self handleSyncResponseIfRequired];
```

Comment:
We need an async return. There is a race condition where we will start a new /sync not using the very last event stream token in `SyncResponseFileStore`.

Reply:
Updated